src/test/kotlin/uk/gov/justice/digital/hmpps/whereabouts/repository/AppointmentLinkerSupportTest.kt

```diff
@@ -0,0 +1,223 @@
+package uk.gov.justice.digital.hmpps.whereabouts.repository
+
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.AfterEach
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest
+import org.springframework.boot.test.mock.mockito.MockBean
+import org.springframework.context.annotation.Import
+import org.springframework.test.context.ActiveProfiles
+import org.springframework.transaction.annotation.Transactional
+import uk.gov.justice.digital.hmpps.whereabouts.model.HearingType
+import uk.gov.justice.digital.hmpps.whereabouts.model.VideoLinkAppointment
+import uk.gov.justice.digital.hmpps.whereabouts.model.VideoLinkBooking
+import uk.gov.justice.digital.hmpps.whereabouts.security.AuthenticationFacade
+
+@ActiveProfiles("test")
+@Import(TestAuditConfiguration::class)
+@DataJpaTest
+@Transactional
+class AppointmentLinkerSupportTest {
+  @Autowired
+  lateinit var videoLinkAppointmentRepository: VideoLinkAppointmentRepository
+
+  @Autowired
+  lateinit var videoLinkBookingRepository: VideoLinkBookingRepository
+
+  @MockBean
+  lateinit var authenticationFacade: AuthenticationFacade
+
+  @BeforeEach
+  fun setup() {
+    clearRepositories()
+    insertTestData()
+  }
+
+  @AfterEach
+  fun tearDown() {
+    clearRepositories()
+  }
+
+  private fun clearRepositories() {
+    videoLinkBookingRepository.deleteAll()
+    videoLinkAppointmentRepository.deleteAll()
+  }
+
+  /**
+   * Creates appointments for 5 booking ids.
+   * bookingId 1 has one linked and one unlinked MAIN appointment.
+   * bookingId 2 has one linked MAIN, PRE and POST appointment, two unlinked PRE and two unlinked POST appointment.
+   * bookingId 3 has two unlinked MAIN appointments. bookingId 3 should be in the result.
+   * bookingId 4 has one unlinked PRE appointment. bookingId 4 should not be in the result.
+   * bookingId 5 has one unlinked POST appointment. bookingId 4 should not be in the result.
+   */
+
+  private fun insertTestData() {
+    var appointmentIdCounter = 100L
+
+    videoLinkBookingRepository.save(
```

Comment:
This would be easier to read if the formatter didn't insist on folding on to multiple lines.

Reply:
Yes I also dislike that rule and considered suggesting we turn it off.  However I like everything else the formatter does and I'm not sure we want to get into the game of arguing over our own ruleset so I decided to suck it up.